va/va_prot.h

```diff
@@ -0,0 +1,404 @@
+ *     }
+ *     vaBeginPicture(va_dpy, decode_ctx, surface);
+ *     ...
+ *     vaRenderPicture(va_dpy, decode_ctx, &buf_id1, 1);
+ *     vaRenderPicture(va_dpy, decode_ctx, &buf_id2, 1);
+ *     ...
+ *     vaEndPicture(va_dpy, decode_ctx);
+ *     if (frame_encrypted) {
+ *         vaDetachProtectedSession(va_dpy, decode_ctx);
+ *     }
+ * }
+ * \endcode
+ *
+ * Or we can attach/detach for every frame
+ * \code
+ * vaAttachProtectedSession(va_dpy, decode_ctx, protected_session);
+ * foreach (iteration) {
+ *     vaBeginPicture(va_dpy, decode_ctx, surface);
+ *     ...
+ *     vaRenderPicture(va_dpy, decode_ctx, &buf_id1, 1);
+ *     vaRenderPicture(va_dpy, decode_ctx, &buf_id2, 1);
+ *     ...
+ *     vaEndPicture(va_dpy, decode_ctx);
+ * }
+ * vaDetachProtectedSession(va_dpy, decode_ctx);
+ * \endcode
+ */
+
+/**
+ * ProtectedSessions and Contexts
+ *
+ * Context represents a "virtual" video decode, encode or video processing
+ * pipeline. Surfaces are render targets for a given context. The data in the
+ * surfaces are not accessible to the client except if derived image is supported
+ * and the internal data format of the surface is implementation specific.
+ *
+ * A protected session is a session between application and GPU hardware.
+ * Protected sessions can be attached to display to enable protection in the pipeline.
+ *
+ * Both contexts and protected sessions are identified by unique IDs and its
+ * implementation specific internals are kept opaque to the clients
+ */
+typedef VAContextID VAProtectedSessionID;
+
+/** \brief Input/Output buffer of VAProtectedSessionExecuteBuffer or VAProtectedSessionHwUpdateBuffer */
+typedef struct _VAProtectedSessionBuffer
+{
+    /*
+     * This is used when this buffer refer to output buffer. The maximum size of data that the driver
+     * can return in the output buffer. It is not used for input buffer.
+     */
+    uint32_t    max_data_size;
+    /*
+     * If it is used for input buffer, it is the size of the input data. If it is used for output buffer,
+     * it is the returns size of the output data written by the driver.
+     */
+    uint32_t    data_size;
+    /*
+     * data pointer of this buffer
+     */
+    void        *data;
+    uint32_t    va_reserved[VA_PADDING_LOW];
+} VAProtectedSessionBuffer;
+
+/** \brief Buffer for vaProtectedSessionExecute() */
+typedef struct _VAProtectedSessionExecuteBuffer {
+    /** \brief Specify the function to execute. It is IHV's implementation specific */
+    uint32_t                    function_id;
+    /** \brief Input buffer */
+    VAProtectedSessionBuffer    input;
+    /** \brief Output buffer */
+    VAProtectedSessionBuffer    output;
+    /** \brief Return the result of this function. It is IHV's implementation specific */
+    uint32_t                    status;
+    uint32_t                    va_reserved[VA_PADDING_LOW];
+} VAProtectedSessionExecutecBuffer;
```

Comment:
Mispelled, rename to VAProtectedSessionExecuteBuffer (unless you are merging this with the VAProtectedSessionHwUpdateBuffer).

Reply:
Updated, thanks for this catch.